### ISSUE TITLE
Docker-compose.yml on Synology

### ISSUE THREAD

**Issue by Anas92230** [espebra/filebin2#61]
Hello,

I try to run docker-compose.yml on my synology. All it's ok on local.
With my domain name:
-App is ok
-Upload is ok
-Download is KO. Link is : https://storage:9000...
Someone can help me to solve this point :-)

Thank you

**Comment by espebra** [comment]
The s3 endpoint `storage:9000` is used in the docker compose file for the local development environment. For access to s3 for external hosts, you need to specify the external address to the Minio service as the s3 endpoint.

The s3 endpoint is set using the `S3_ENDPOINT` environment variable in `docker-compose.yml` or the `--s3-endpoint` command line parameter to the filebin application.

Can you try this?

**Comment by Anas92230** [comment]
Hello @espebra,
like this:
S3_ENDPOINT=https://storage.domain.com:9000 ? 
``
version: "3"
services:
  s3:
    image: "minio/minio"
    hostname: "storage"
    restart: "no"
    environment:
      - MINIO_ROOT_USER=admin
      - MINIO_ROOT_PASSWORD=passwd
    expose:
      - "9000"
      - "9001"
    ports:
      - "9000:9000"
      - "9001:9001"
    entrypoint: [ "minio", "server", "/data", "--console-address", ":9001" ]
    networks:
      - local

  db:
...
  app:
    build: .
    restart: "no"
    environment:
      - DATABASE_HOST=db
      - DATABASE_PORT=5432
      - DATABASE_NAME=db
      - DATABASE_USERNAME=admin
      - DATABASE_PASSWORD=passwd
      **- S3_ENDPOINT=https://storage.domain.com:9000**
      - S3_REGION=us-east-1
      - S3_BUCKET=filebin
      - S3_ACCESS_KEY=admin
      - S3_SECRET_KEY=passwd
      - ADMIN_USERNAME=admin
      - ADMIN_PASSWORD=passwd
      - METRICS_USERNAME=admin
      - METRICS_PASSWORD=passwd
      - METRICS_AUTH=basic
    expose:
      - "8080"
    ports:
      - "8080:8080"
    volumes:
      - .:/app:delegated
    depends_on:
      - db
      - s3
    networks:
      - local
    entrypoint: [ "./wait-for-s3.sh", "make", "run" ]
networks:
  local: null
``

**Comment by espebra** [comment]
That should be it, yes. In addition, that port (9000) needs to be exposed externally from the s3 container.

**Comment by Anas92230** [comment]
thanks I try it but no result :-( I have this error for app:
-Unable to initialize S3 connection: Endpoint url cannot have fully qualified paths.
minio is accessible from local. If I want to share, here is the link : http://127.0.0.1:9000/filebin/blablabla

**Comment by Anas92230** [comment]
other info:
if I change this url by https://storage.mydomain.com/blablabla I can download file.

**Comment by espebra** [comment]
Yes, that makes sense. You need to specify the external endpoint to S3 when specifying the S3 endpoint in Filebin. Note that archive downloads should work independent from this, since archive downloads are proxied via the Filebin app.

I believe the issue is solved. Correct?

**Comment by Anas92230** [comment]
No. this parameter S3_ENDPOINT don't work :-(
Do you ever try it for external use?

**Comment by espebra** [comment]
I only use the docker compose file as it is in the repository, for local testing. In production I use the `S3_ENDPOINT` environment variable pointing to the external hostname of the S3 service.

**Comment by espebra** [comment]
Can you please try to update the [wait-for-s3.sh](https://github.com/espebra/filebin2/blob/master/wait-for-s3.sh) script so that it is using the external hostname for S3 as well?

**Comment by Anas92230** [comment]
Hello,
ok I'm going to try it.
Thank you

**Comment by Anas92230** [comment]
Hello,

I successed in installation with Minio only. All it's ok from external I can upload file and share it with someone and it's possible for him to download it.
here's my docker-compose : 
`version: "3.8"

services:
  minio:
    image: quay.io/minio/minio
    command: server /data --console-address ":9001"
    volumes:
      - minio_data:/volume1/docker/minio/data
    ports:
      - 9090:9000
      - 9001:9001
    environment:
      MINIO_ROOT_USER: username
      MINIO_ROOT_PASSWORD: password
      MINIO_BROWSER_REDIRECT_URL: https://console.mydomain.com
      MINIO_SERVER_URL: https://s3.mydomain.com

volumes:
  minio_data:
    external: true`
But no success with filebin2:
**docker logs filebin2-app-1 :** 
_S3 service is available.
go version
go version go1.20.5 linux/amd64
mkdir -p artifacts tests
GOOS=linux GOARCH=amd64 go build -mod=vendor -o artifacts/filebin2-linux-amd64 -trimpath -buildvcs=false
artifacts/filebin2-linux-amd64 --listen-host 0.0.0.0 --lurker-interval 10 --expiration 3600 --access-log=access.log --s3-secure=false --db-host=db --limit-storage 1G --admin-username admin --admin-password changeme --metrics
Unable to load geoip database: open : no such file or directory
TTL for presigned S3 URLs: 1m0s
**Unable to initialize S3 connection: Endpoint url cannot have fully qualified paths.**
make: *** [Makefile:18: run] Error 2_

**Comment by espebra** [comment]
The error message says that the (s3) endpoint URL can't be a fully qualified path. Can you try specifying `console.mydomain.com` instead of https://console.mydomain.com/? It will by default expect https:// and port 443.

**Comment by Anas92230** [comment]
i change only in filebin service. I have an other issue:
docker logs filebin2-app-1
S3 service is available.
go version
go version go1.20.5 linux/amd64
mkdir -p artifacts tests
GOOS=linux GOARCH=amd64 go build -mod=vendor -o artifacts/filebin2-linux-amd64 -trimpath -buildvcs=false
artifacts/filebin2-linux-amd64 --listen-host 0.0.0.0 --lurker-interval 10 --expiration 3600 --access-log=access.log --s3-secure=false --db-host=db --limit-storage 1G --admin-username admin --admin-password changeme --metrics
Unable to load geoip database: open : no such file or directory
TTL for presigned S3 URLs: 1m0s
**Established session to S3AO at s3.mydomain.com
Unable to check if S3AO bucket exists: Access Denied.
Unable to initialize S3 connection: Access Denied.**
make: *** [Makefile:18: run] Error 2

**Comment by Anas92230** [comment]
another info if I change value of S3_ENDPOINT with storage:9000, I have in log: 
docker logs filebin2-app-1
S3 service is available.
go version
go version go1.20.5 linux/amd64
mkdir -p artifacts tests
GOOS=linux GOARCH=amd64 go build -mod=vendor -o artifacts/filebin2-linux-amd64 -trimpath -buildvcs=false
artifacts/filebin2-linux-amd64 --listen-host 0.0.0.0 --lurker-interval 10 --expiration 3600 --access-log=access.log --s3-secure=false --db-host=db --limit-storage 1G --admin-username admin --admin-password changeme --metrics
Unable to load geoip database: open : no such file or directory
TTL for presigned S3 URLs: 1m0s
Established session to S3AO at storage:9000
**Unable to check if S3AO bucket exists: Get "http://storage:9000/filebin/?location=": dial tcp: lookup storage on 127.0.0.11:53: no such host
Unable to initialize S3 connection: Get "http://storage:9000/filebin/?location=": dial tcp: lookup storage on 127.0.0.11:53: no such host**
make: *** [Makefile:18: run] Error 2

**Comment by espebra** [comment]
It took a while, but I think this can be closed now - both because it is old but also because I think filebin is easier to configure now.

* Filebin is now released as a distroless container image on Github.
* Everything is configurable as environment variables, which means that an entrypoint script is not needed.

The various ports are configured only when the application starts and not in any scripts like wait-for-s3.sh.

The relevant documentation is [getting started with docker compose or docker image](https://github.com/espebra/filebin2?tab=readme-ov-file#getting-started) and [list of environment variables](https://github.com/espebra/filebin2?tab=readme-ov-file#configuration).
